src/app/modules/group/components/pending-request/pending-request.component.ts

```diff
@@ -154,30 +162,42 @@ export class PendingRequestComponent implements OnInit, OnChanges {
   }
 
   onAcceptOrReject(action: Action): void {
-    if (this.selection.length === 0 || this.ongoingActivity !== 'none') {
+    if (this.selection.length === 0 || this.state !== 'ready') {
       return;
     }
-    this.ongoingActivity = (action === 'accept') ? 'accepting' : 'rejecting';
-
-    const resultObserver : Observable<Map<string, any>[]> = this.processRequests(action);
-
-    resultObserver
-      .subscribe(
-        res => {
-          this.displayResponseToast(
-            this.parseResults(res),
-            action === 'accept' ? 'accept' : 'reject', // still use a matching as it is "by coincidence" that the type of verb match
-            action === 'accept' ? 'accepted' : 'declined'
-          );
-          this.reloadData();
-          this.ongoingActivity = 'none';
-          this.selection = [];
-        },
-        err => {
-          this.processRequestError(err);
-          this.ongoingActivity = 'none';
-        }
-      );
+
+    if (action === 'accept') this.state = 'accepting';
+    else this.state = 'rejecting';
+
+    this.processRequests(action, this.selection).pipe(
+      map(result => ({
+        result: result,
+        action: action,
+        refreshData: { groupId: this.groupId, includeSubgroup: this.includeSubgroup, sort: this.currentSort },
+      }))
+    ).subscribe(
+      state => {
+        this.state = 'ready';
```

Comment:
This function and the switchMap on the subject conflicts a bit on updating the state. So it should be ok, but it is not really so straightforward to prove that there is no way to have a conflict in one scenario.
But that's ok :-)